ui/app/components/console/ui-panel.js

```diff
@@ -0,0 +1,96 @@
+import Ember from 'ember';
+import {
+  parseCommand,
+  extractDataAndFlags,
+  logFromResponse,
+  logFromError,
+  logErrorFromInput,
+} from 'vault/lib/console-helpers';
+
+const { inject, computed } = Ember;
+
+export default Ember.Component.extend({
+  classNames: 'console-ui-panel-scoller',
+  classNameBindings: ['isFullscreen:fullscreen'],
+  isFullscreen: false,
+  console: inject.service(),
+  inputValue: null,
+  log: computed.alias('console.log'),
+
+  logAndOutput(command, logContent) {
+    this.set('inputValue', '');
+    this.get('console').logAndOutput(command, logContent);
+  },
+
+  executeCommand(command, shouldThrow = false) {
+    let service = this.get('console');
+    let serviceArgs;
+    if (command === 'clearall') {
+      this.logAndOutput(command);
+      service.clearLog(true);
+      return;
+    }
+    if (command === 'clear') {
+      this.logAndOutput(command);
+      service.clearLog();
+      return;
+    }
+    if (command === 'fullscreen') {
+      this.toggleProperty('isFullscreen');
+      this.logAndOutput(command);
+      return;
+    }
+    // parse to verify it's valid
+    try {
+      serviceArgs = parseCommand(command, shouldThrow);
+    } catch (e) {
+      this.logAndOutput(command, { type: 'help' });
+      return;
+    }
+    // we have a invalid command but don't want to throw
+    if (serviceArgs === false) {
+      return;
+    }
+
+    let [method, flagArray, path, dataArray] = serviceArgs;
+
+    if (dataArray || flagArray) {
+      var { data, flags } = extractDataAndFlags(dataArray, flagArray);
+    }
+
+    let inputError = logErrorFromInput(path, method, flags, dataArray);
+    if (inputError) {
+      this.logAndOutput(command, inputError);
+      return;
+    }
+    let serviceFn = service[method];
+    serviceFn.call(service, path, data, flags.wrapTTL)
+      .then(resp => {
+        this.logAndOutput(command, logFromResponse(resp, path, method, flags));
+      })
+      .catch(error => {
+        this.logAndOutput(command, logFromError(error, path, method));
+      });
```

Comment:
Any time I see `.then` in a component, I think `ember-concurrency`. Normally, you'd have to do an `isDestroyed(this)` check here, but because this component happens to be always-rendered at the application level, you can get away without it.

Still, maybe it's a good practice to always use an e-c task when you're awaiting an async action and performing some component state manipulation when it's complete? That way if the component becomes conditionally rendered in the future, you won't see exceptions thrown.